mesa_frames/concrete/datacollector.py

```
@@ -177,13 +177,94 @@ def _collect_agent_reporters(self, current_model_step: int, batch_id: int):
+                        for key, val in reg_result.items():
+                            if isinstance(val, pl.Series):
+                                agent_data_dict[f"{col_name}_{key}"] = val
+                            elif isinstance(val, pl.DataFrame):
+                                for subcol in val.columns:
+                                    agent_data_dict[f"{col_name}_{key}_{subcol}"] = val[
+                                        subcol
+                                    ]
+                            else:
+                                raise TypeError(
+                                    "Registry-level reporter dict values must be Series or DataFrame"
+                                )
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)